src/database.ts

```diff
@@ -55,24 +60,104 @@ import {
   TransactionRunner,
 } from './transaction-runner';
 
-export interface GetDatabaseOptions {
-  autoCreate?: boolean;
-}
-export type DatabaseResponse = [Database, r.Response];
-export interface DatabaseCallback {
-  (err: Error | null, database?: Database, apiResponse?: r.Response): void;
-}
+import {google} from '../proto/spanner';
+import {
+  Schema,
+  RequestCallback,
+  PagedRequest,
+  ResourceCallback,
+  PagedResponse,
+} from './common';
+import {ServiceError, CallOptions, ClientDuplexStream} from 'grpc';
+import {Readable, Transform} from 'stream';
+import {PreciseDate} from '@google-cloud/precise-date';
+import {google as spannerClient} from '../proto/spanner';
+import {RequestConfig} from '.';
+
+type CreateBatchTransactionCallback = ResourceCallback<
+  BatchTransaction,
+  google.spanner.v1.ITransaction | google.spanner.v1.ISession
+>;
+
+type CreateBatchTransactionResponse = [
+  BatchTransaction,
+  google.spanner.v1.ITransaction | google.spanner.v1.ISession
+];
+type DatabaseResponse = [Database, r.Response];
+type DatabaseCallback = ResourceCallback<Database, r.Response>;
+
+type GetSnapshotCallback = ResourceCallback<Snapshot>;
```

Comment:
Should we include the response type?

Reply:
`GetSnapshotCallback` has only two argument as Error and Snapshot.

Reply:
Ah, my mistake.